code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity/IdentityECIDHandlingTest.java

```diff
@@ -117,10 +117,11 @@ public void testECID_onResetClearsOldECID() throws Exception {
 	public void testECID_AreDifferentAfterPrivacyChange() throws Exception {
 		/// Test Edge Identity and IdentityDirect have same ECID on bootup, and after privacy change ECIDs are different
 		registerIdentityDirectExtension();
+		TestHelper.waitForThreads(1000);
 
 		// register EdgeIdentity extension
 		Identity.registerExtension();
-		TestHelper.waitForThreads(2000);
+		TestHelper.waitForThreads(1000);
```

Comment:
Should we use registerBothIdentityExtensions instead?

Reply:
If we register both extensions at the same time, the direct ECID will become secondary ECID for edgeIdentity extension and when privacy toggle happens it will change only the secondary ECID.

Reply:
I think this test needs a few adjustments if we are testing that at first boot it reads the identity direct ecid and on privacy toggle it resets only secondary. 
1. set the persistence for Identity direct
2. register both
3. check same ecid
4. privacy toggle
5. check different ecids

Reply:
And this test case just wants to make sure that the ECID is refreshed at EdgeIdentity Extension on toggling privacy change and hence we can still do  registerBothIdentityExtensions and see if the secondary ECID is refreshed


```
    @Test
    public void testECID_AreDifferentAfterPrivacyChange() throws Exception {
        /// Test Edge Identity and IdentityDirect have same ECID on bootup, and after privacy change ECIDs are different
        registerBothIdentityExtensions();

        String edgeECID = getExperienceCloudIdSync();
        String oldDirectECID = getIdentityDirectECIDSync();

        // verify ECID's in edge
        verifyPrimaryECID(edgeECID);
        verifySecondaryECID(oldDirectECID);

        //  Toggle privacy
        togglePrivacyStatus();

        // verify legacy ECID added to IdentityMap and it is changed
        String newDirectECID = getIdentityDirectECIDSync();
        assertNotNull(newDirectECID);
        assertNotEquals(newDirectECID, oldDirectECID);
        verifyPrimaryECID(edgeECID);
        verifySecondaryECID(newDirectECID);
    }

```